instrumentation/net/http/otelhttp/common.go

```diff
@@ -31,10 +31,16 @@ const (
 
 // Server HTTP metrics.
 const (
-	RequestCount          = "http.server.request_count"           // Incoming request count total
-	RequestContentLength  = "http.server.request_content_length"  // Incoming request bytes total
-	ResponseContentLength = "http.server.response_content_length" // Incoming response bytes total
-	ServerLatency         = "http.server.duration"                // Incoming end to end duration, milliseconds
+	serverRequestSize  = "http.server.request.size"  // Incoming request bytes total
+	serverResponseSize = "http.server.response.size" // Incoming response bytes total
+	serverDuration     = "http.server.duration"      // Incoming end to end duration, milliseconds
```

Comment:
`go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` is at `v0.46.1`. If now is not the right moment to make breaking changes, then... when is? I may be wrong, but I thought breaking changes are ok and had happened in `v0.x.x` modules here before. 🤔

Reply:
I originally tried not to include any breaking changes but got guidance from @pellared to update these: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4707#issuecomment-1871335266

Reply:
@MadVikingGod

Reply:
Yes, we can make breaking changes, but no, we shouldn't do this lightly.

We need to deprecate these constants before we pull them out.  Pulling them out can make some user's code incompatible with newer versions of this package (and potentially the broader ecosystem).  While I do think that there is a low chance this will break any of the [827](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp) that import this library, we should act carefully around breaking changes.

Reply:
Added back and added docs about deprecation here: 70c1791e8dbdcd7324c5b1c78645b3e4a2b1d695